9. Graph/task_prerequisite/task_prerequisite_harikrishnan.cpp

```diff
@@ -37,11 +37,16 @@ bool checkCycle(int n, vector<int> adj[], vector<bool>& vis, vector<bool>& rec)
     rec[n] = false;  // Remove the node from the recursive stack. 
     return false;   
 }
- 
+
 void solve()
 {
     int N,K,u,v;
     cin>>N>>K;                 // Input the string
+    if(N == 0)            // No tasks gievn.
```

Comment:
You only added this snippet of code and this is not enough for the logic to be correct.

Reply:
For the input
2 1
0 1
Expected output is YES